elasticsearch/src/main/java/io/micronaut/elasticsearch/DefaultElasticsearchClientFactory.java

```diff
@@ -54,6 +63,18 @@ RestClient restClient(DefaultElasticsearchConfigurationProperties elasticsearchC
         return restClientBuilder(elasticsearchConfiguration).build();
     }
 
+    /**
+     * @param elasticsearchConfiguration The {@link DefaultElasticsearchConfigurationProperties} object
+     * @return The Elasticsearch Client
+     */
+    @Singleton
+    ElasticsearchClient elasticsearchClient(DefaultElasticsearchConfigurationProperties elasticsearchConfiguration, ObjectMapper objectMapper) {
+      RestClient restClient = restClientBuilder(elasticsearchConfiguration).build();
+
+      ElasticsearchTransport transport = new RestClientTransport(restClient, new JacksonJsonpMapper(objectMapper));
```

Comment:
Long term this shouldn't be coupled Jackson. Seems this  interface needs to be implemented https://artifacts.elastic.co/javadoc/co/elastic/clients/elasticsearch-java/7.16.2/co/elastic/clients/json/JsonpMapper.html

However perhaps that is not possible for now and can / should be addressed later.

Reply:
@yawkat please file a ticket to track implementing this for elastic search

Reply:
It would be a big refactoring right now, I suggest doing this at a later stage.